docs/scenarios_authoring/RRDataSourcesAndSinks.md

```diff
@@ -2,42 +2,73 @@
 sidebar_position: 4
 ---
 
-# Request-Response sources and sinks
+# Request-Response specific components
 
-## Http request as input and output to Nussknacker scenarios
+## Concepts
 
-### Concepts
-
-In Request-Response mode you interact with Nussknacker Lite runtime engine in a similar way as with any "server" in the
+### General
+In Request-Response mode you interact with Nussknacker Lite runtime engine in the same way as with any "server" in the
 Internet. You send request which contains request parameters and you get response from the server. The protocol for this
-communication is HTTP - the same protocol is used by browsers when they communicate with web servers. One of the
-differences is that the content of the communication with the Nussknacker Lite engine ("the server") is encoded as JSON,
-not HTML. Not very surprisingly, in Request-Response processing mode the only possible Source component is Request and
-the only Sink component is Response.
+communication is HTTP - the same protocol is used by browsers when they communicate with web servers. When the  browser is used to browse internet pages, the web server responds with content encoded as HTML. In Nussknacker case, the Nussknacker Lite engine ("the server") respondes with content encoded as JSON. You may envoke Nussknacker Lite engine directly from the browser and watch the JSON response in the browser window. 
 
-## Request-Response schema
+Not very surprisingly, in the Request-Response processing mode the only possible source component is request and the only sink component is response.
 
-Each scenario needs to define two schemas in [JSON Schema](https://json-schema.org/) format:
+### Schemas
+Unlike in the streaming processing mode, there is no Schema Registry which store data schemas - request and response schemas in our case. Still, the data schema of the request and response ("input" and "output") need to be declared to Nussknacker; they are defined alongside the scenario (as the scenario property). The "language" used to define schemas is called [JSON Schema](https://json-schema.org/). Nussknacker uses those schemas to understand how the request and response should look like, assist you with hints and validate expressions you write. 
```

Comment:
Will it be understandable for those who didn't read streaming description?

Reply:
The idea here is only to highlight the fact that there is a difference against streaming processing mode. Understanding this information does not require any prior or additional understanding of the streaming mode, it is just a fact which is easy to accept.